src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -76,4 +97,58 @@ class WorkspaceService(protected val argUserInfo: UserInfo, val rawlsDAO: RawlsD
   }
 
+  def exportWorkspaceAttributesTSV(workspaceNamespace: String, workspaceName: String, filename: String): Future[PerRequestMessage] = {
+    rawlsDAO.getWorkspace(workspaceNamespace, workspaceName) map { workspaceResponse =>
+      val attributes = workspaceResponse.workspace.get.attributes.filterKeys(_ != AttributeName("default", "description"))
+      val headerString = "workspace:" + (attributes map { case (attName, attValue) => attName.name }).mkString("\t")
+      val valueString = (attributes map { case (attName, attValue) => impPlainAttributeFormat.write(attValue) }).mkString("\t")
+      RequestCompleteWithHeaders((StatusCodes.OK, headerString + "\n" + valueString),
+        HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> filename)),
+        HttpHeaders.`Content-Type`(`text/plain`))
+    }
+  }
+
+  def importAttributesFromTSV(workspaceNamespace: String, workspaceName: String, tsvString: String): Future[PerRequestMessage] = {
+    Try(TSVParser.parse(tsvString)) match {
+      case Failure(regret) => Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, regret.getMessage))
+      case Success(tsv) =>
+        tsv.firstColumnHeader.split(":")(0) match {
+          case "workspace" =>
+            importWorkspaceAttributeTSV(workspaceNamespace, workspaceName, tsv)
+          case _ =>
+            Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid first column header should start with \"workspace\""))
+        }
+    }
+  }
+
+  private def importWorkspaceAttributeTSV(workspaceNamespace: String, workspaceName: String, tsv: TSVLoadFile): Future[PerRequestMessage] = {
+    if (tsv.tsvData.length != 1) {
+      Future(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Your file does not have the correct number of rows. There should be 2."))
+    } else {
+      val attributeNames = Seq(tsv.headers.head.stripPrefix("workspace:")) ++ tsv.headers.tail
+      val distinctAttributes = attributeNames.distinct
+      if (attributeNames.size != distinctAttributes.size) {
+        Future(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Duplicated attribute keys are not allowed"))
+      } else {
+        rawlsDAO.getWorkspace(workspaceNamespace, workspaceName) flatMap { workspaceResponse =>
+          Try(getWorkspaceAttributeCalls(attributeNames.zip(tsv.tsvData.head))) match {
+            case Failure(regret) => Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest,
+              "One or more of your values are not in the correct format"))
+            case Success(attributeCalls) => rawlsDAO.patchWorkspaceAttributes(workspaceNamespace, workspaceName, attributeCalls) map (RequestComplete(_))
+          }
+        }
+      }
+    }
+  }
+
+  private def getWorkspaceAttributeCalls(attributePairs: Seq[(String,String)]): Seq[AttributeUpdateOperation] = {
+    attributePairs.map { case (name, value) =>
+      if (value.equals("__DELETE__"))
```

Comment:
This definitely exists for entities. We should make more of an effort to share this code rather than duplicate it.